pcsx2/GS/Renderers/HW/GSRendererNew.cpp

```diff
@@ -760,6 +746,20 @@ void GSRendererNew::EmulateBlending(bool& DATE_GL42, bool& DATE_GL45)
 			m_conf.blend = {blend_index, ALPHA.FIX, ALPHA.C == 2, false, false};
 		}
 	}
+
+	// GL42 interact very badly with sw blending. GL42 uses the primitiveID to find the primitive
+	// that write the bad alpha value. Sw blending will force the draw to run primitive by primitive
+	// (therefore primitiveID will be constant to 1).
+	// Switch DATE_GL42 with DATE_GL45 in such cases to ensure accuracy.
+	// No mix of COLCLIP + sw blend + DATE_GL42, neither sw fbmask + DATE_GL42.
+	// Note: Do the swap after colclip to avoid adding extra conditions.
+	if (sw_blending && DATE_GL42 && m_conf.require_full_barrier)
+	{
+		GL_PERF("DATE: Swap DATE_GL42 with DATE_GL45");
+		m_conf.require_full_barrier = true;
```

Comment:
Why set this to true when it needs to be true to get in here in the first place? :)

Reply:
Because I cut/pasted it :D